packages/messaging/src/testing/compare-headers.ts

```diff
@@ -0,0 +1,38 @@
+/**
+ * @license
+ * Copyright 2019 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { expect } from 'chai';
+import '../testing/setup';
+
+// Trick TS since it's set to target ES5.
+declare class HeadersWithEntries extends Headers {
+  entries?(): Iterable<[string, string]>;
+}
+
+// Chai doesn't check if Headers objects contain the same entries,
+// so we need to do that manually.
+export function compareHeaders(
+  expectedHeaders: HeadersWithEntries,
+  actualHeaders: HeadersWithEntries
+): void {
+  expect(expectedHeaders.entries).not.to.be.undefined;
+  expect(actualHeaders.entries).not.to.be.undefined;
+
+  const expected = new Map(Array.from(expectedHeaders.entries!()));
```

Comment:
Doesn't the Map constructor take an iterable why do you need Array.from()?

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map

Reply:
It worked, so yes it does. :smile: